### PR TITLE
[Feature] Support mmseg with NPU backend.

### PR DESCRIPTION
**Pull request by luomaoling**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Added ascending device support in mmseg.

## Modification

The main modification points are as follows:
   We added an NPU device in the DDP scenario and DP scenario when using the NPU.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.
None

## Use cases (Optional)

We tested [fcn_unet_s5-d16_4x4_512x1024_160k_cityscapes.py](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/unet/fcn_unet_s5-d16_4x4_512x1024_160k_cityscapes.py) .

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by xiexinch** [comment]
Hi @luomaoling,
Thanks for your contribution! Please sign the CLA.

File: mmseg/utils/util_distribution.py [hunk 1]
```diff
@@ -33,6 +33,14 @@ def build_dp(model, device='cuda', dim=0, *args, **kwargs):
         dp_factory['mlu'] = MLUDataParallel
         model = model.mlu()
 
+    if device == 'npu':
```
Comment (xiexinch) [inline mmseg/utils/util_distribution.py:36]: Might add the `elif` statement to the code above

File: mmseg/utils/util_distribution.py [hunk 2]
```diff
@@ -53,7 +61,8 @@ def build_ddp(model, device='cuda', *args, **kwargs):
         .. [1] https://pytorch.org/docs/stable/generated/torch.nn.parallel.
                      DistributedDataParallel.html
     """
-    assert device in ['cuda', 'mlu'], 'Only available for cuda or mlu devices.'
+    assert device in ['cuda', 'mlu', 'npu'], 'Only available for cuda '\
+                                             'or mlu devices.'
```
Comment (xiexinch) [inline mmseg/utils/util_distribution.py:65]: Might add `npu` to the error msg.

File: mmseg/utils/util_distribution.py [hunk 3]
```diff
@@ -63,6 +72,14 @@ def build_ddp(model, device='cuda', *args, **kwargs):
         ddp_factory['mlu'] = MLUDistributedDataParallel
         model = model.mlu()
 
+    if device == 'npu':
```
Comment (xiexinch) [inline mmseg/utils/util_distribution.py:75]: Might use `elif` statement.

**Review by xiexinch — APPROVED** [review]

**Review by MengzhangLI — APPROVED** [review]

File: tools/test.py [hunk 4]
```diff
@@ -260,6 +260,12 @@ def main():
         tmpdir = None
 
     cfg.device = get_device()
+
+    if cfg.device == 'npu':
+        optimiter_config = dict(type='Fp16OptimizerHook', loss_scale='dynamic')
+        cfg.optimizer_config = optimiter_config if \
+            not cfg.optimizer_config else cfg.optimizer_config
```
Comment (MeowZheng) [inline tools/test.py:267]: when model testing, there is no runner and no hooks

Comment (luomaoling) [reply tools/test.py]: I've revised it, thanks！

File: mmseg/utils/util_distribution.py (hunk 3 above)
Comment (luomaoling) [reply mmseg/utils/util_distribution.py]: I've revised it, thanks

File: mmseg/utils/util_distribution.py (hunk 2 above)
Comment (luomaoling) [reply mmseg/utils/util_distribution.py]: I've revised it, thanks

File: mmseg/utils/util_distribution.py (hunk 1 above)
Comment (luomaoling) [reply mmseg/utils/util_distribution.py]: I've revised it, thanks

**Pull request merged**